extensions/applicationinsights-react-native/src/ReactNativePlugin.ts

```diff
@@ -9,149 +9,170 @@ import {
     IPlugin,
     IAppInsightsCore, 
     CoreUtils,
-    IDiagnosticLogger,
     LoggingSeverity,
     _InternalMessageId,
-    DiagnosticLogger
+    BaseTelemetryPlugin,
+    IProcessTelemetryContext
 } from '@microsoft/applicationinsights-core-js';
 import { ConfigurationManager, IDevice, IExceptionTelemetry, IAppInsights, SeverityLevel, AnalyticsPluginIdentifier  } from '@microsoft/applicationinsights-common';
 import DeviceInfo from 'react-native-device-info';
 
 import { INativeDevice, IReactNativePluginConfig } from './Interfaces';
+import dynamicProto from '@microsoft/dynamicproto-js';
 
-export class ReactNativePlugin implements ITelemetryPlugin {
+export class ReactNativePlugin extends BaseTelemetryPlugin {
```

Comment:
Upgraded to use the new base class, so that it will properly support being used in a shared environment (reusing the same plugin from multiple app insight instances)